packages/office-ui-fabric-react/src/components/Announced/examples/Announced.QuickActions.Example.tsx

```diff
@@ -53,177 +52,162 @@ export interface IAnnouncedQuickActionsExampleItem {
 }
 
 export interface IAnnouncedQuickActionsExampleState {
-  items: IAnnouncedQuickActionsExampleItem[];
-  renameDialogOpen: boolean;
-  dialogContent: JSX.Element | undefined;
-  announced?: JSX.Element;
+  selection: Selection;
 }
 
-export class AnnouncedQuickActionsExample extends React.Component<{}, IAnnouncedQuickActionsExampleState> {
-  private _selection: Selection;
-  private _detailsList = React.createRef<IDetailsList>();
-  private _textField = React.createRef<ITextField>();
-  private _async: Async;
+export const AnnouncedQuickActionsExample: React.FunctionComponent = () => {
+  const detailsList = React.useRef<IDetailsList>(null);
+  const textField = React.useRef<ITextField>(null);
+  const selection = useConst(() => new Selection());
+  const [items, setItems] = React.useState<IAnnouncedQuickActionsExampleItem[]>(() => {
+    const itemsList: IAnnouncedQuickActionsExampleItem[] = [];
+    for (let i = 0; i < 20; i++) {
+      itemsList.push({
+        key: i,
+        name: 'Item ' + i,
+        modified: getMockDateString(),
+        modifiedby: names[Math.floor(Math.random() * names.length)],
+        filesize: Math.floor(Math.random() * 30).toString() + ' MB',
+      });
+    }
+    return itemsList;
+  });
+  const [renameDialogOpen, setRenameDialogOpen] = React.useState<boolean>(false);
```

Comment:
Thought about this part a bit more and you could potentially get rid of this boolean and just check `!!dialogContent` instead (and set `dialogContent` back to undefined when closed)